sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -115,46 +141,193 @@ export class BatchingReceiver extends MessageReceiver {
+ *
+ * @internal
+ * @ignore
+ */
+export class BatchingReceiverLite {
+  constructor(
+    clientEntityContext: ClientEntityContext,
+    private _getCurrentReceiver: (abortSignal?: AbortSignalLike) => Promise<MinimalReceiver>,
+    private _receiveMode: ReceiveMode
+  ) {
+    this._createServiceBusMessage = (context: MessageAndDelivery) => {
+      return new ServiceBusMessageImpl(
+        clientEntityContext,
+        context.message!,
+        context.delivery!,
+        true
+      );
+    };
+
+    this._getRemainingWaitTimeInMsFn = (
+      maxWaitTimeInMs: number,
+      maxTimeAfterFirstMessageInMs: number
+    ) => getRemainingWaitTimeInMsFn(maxWaitTimeInMs, maxTimeAfterFirstMessageInMs);
+
+    this.isReceivingMessages = false;
+  }
+
+  private _createServiceBusMessage: (
+    context: Pick<EventContext, "message" | "delivery">
+  ) => ServiceBusMessageImpl;
+
+  private _getRemainingWaitTimeInMsFn: typeof getRemainingWaitTimeInMsFn;
+  private _closeHandler: ((connectionError?: AmqpError | Error) => void) | undefined;
+
+  isReceivingMessages: boolean;
+
+  /**
+   * Receives a set of messages,
+   *
+   * @internal
+   * @ignore
+   */
+  public async receiveMessages(args: ReceiveMessageArgs): Promise<ServiceBusMessageImpl[]> {
+    try {
+      this.isReceivingMessages = true;
+      const receiver = await this._getCurrentReceiver(args.userAbortSignal);
+      return await this._receiveMessagesImpl(receiver, args);
     } finally {
       this.isReceivingMessages = false;
-      this._connectionErrorHandler = undefined;
     }
   }
 
-  private _receiveImpl(
-    maxMessageCount: number,
-    maxWaitTimeInMs: number,
-    maxTimeAfterFirstMessageMs: number,
-    abortSignal?: AbortSignalLike
+  /**
+   * Closes the receiver (optionally with an error), cancelling any current operations.
+   *
+   * @param connectionError An optional error (rhea doesn't always deliver one for certain disconnection events)
+   */
+  close(connectionError?: Error | AmqpError) {
+    if (this._closeHandler) {
+      this._closeHandler(connectionError);
+      this._closeHandler = undefined;
+    }
+  }
+
+  private _receiveMessagesImpl(
+    receiver: MinimalReceiver,
+    args: ReceiveMessageArgs
   ): Promise<ServiceBusMessageImpl[]> {
     const getRemainingWaitTimeInMs = this._getRemainingWaitTimeInMsFn(
-      maxWaitTimeInMs,
-      maxTimeAfterFirstMessageMs
+      args.maxWaitTimeInMs,
+      args.maxTimeAfterFirstMessageInMs
     );
 
     const brokeredMessages: ServiceBusMessageImpl[] = [];
+    const loggingPrefix = `[${receiver.connection.id}|r:${receiver.name}`;
 
-    this.isReceivingMessages = true;
     return new Promise<ServiceBusMessageImpl[]>((resolve, reject) => {
       let totalWaitTimer: NodeJS.Timer | undefined;
 
       // eslint-disable-next-line prefer-const
       let cleanupBeforeResolveOrReject: (
-        receiver: Receiver | undefined,
         shouldRemoveDrain: "removeDrainHandler" | "leaveDrainHandler"
```

Comment:
I admit that this was not introduced in this PR and so am not saying we should do any changes here, but why is this argument not a boolean?

Reply:
This was to make it clear, at the call site, which one was being chosen.